### DISCUSSION TITLE
Document links not (always) working in editmode in multisite setup

### DISCUSSION THREAD

**Discussion by cx-mezz** [pimcore/pimcore#9522] [Q&A / Support] [answered]
This is probably not a bug, i just would like to understand better how links work in edit mode.

To reproduce this, you need a document, serving as a separate site, and a link on that document, pointing to some inner page on that separate site:

1) Install latest pimcore X from the skeleton project
2) Create a new document `child-domain.local`, serving as a separate site, for example with a domain name `child-domain.local`
        
![image](https://user-images.githubusercontent.com/58294213/122888014-a352b500-d36b-11eb-841b-9528bad66476.png)

3) Add a link area brick to the default page template, templates/default/default.html.twig:

    ```
    {{ pimcore_link('link') }}
    ```

4) Create a new page under `child-domain.local` branch, call it `page1` for instance. On the root document of the child-domain.local branch point the link that was created on the previous step to the new page `page1`.
    
![image](https://user-images.githubusercontent.com/58294213/122888547-296efb80-d36c-11eb-8a62-9b5716560a1c.png)

In the edit mode, the url that is generated for this link is http://localhost/child-domain.local/page1 (given that we access the admin dashboard on http://localhost/admin)

When clicked, the link shows an alert saying "Leave site? Changes you made may not be saved." If admin still wants to follow the link and clicks "Leave", the page1 is opened. It doesn't allow to edit content, but that's fine, I think. At least, admin can see that the link is working and correct page is opened.

5) Now, instead of the main domain, login the admin panel on the child domain, i.e http://child-domain.local/admin. In the edit mode, url that is generated for the link on the domain root page, is http://child-domain.local/child-domain.local/page1. When clicked, this url gives 404 error page, obviously because the full path /child-domain.local/page1 exists only when accessing the main site. On the child domain, correct url is just /page1

So my question, is this intentional? I'm a bit confused with this inconsistency: the link works fine when using the admin dashboard on the main domain, and doesn't work on the child domain. 

May be this inconsistency might be fixed by taking in account the current domain name, when generating links in edit and preview modes. That is, instead of /child-domain.local/page1 the link generator would just give /page1, when accessing the site on http://child-domain.local/ domain where the page belongs to.

Or may be links are not supposed to work in editmode at all? I mean, if you see the alert "Leave site? Changes you made may not be saved." in edit mode, just don't follow the link, because edit mode is supposed to be used to only edit pages, to navigate between them you must use the document tree pane (I would totally understand if that is the case, it just seemed so convenient to me in the single domain mode to click on links in edit/preview mode to ensure that they work).

**Comment by dvesh3** [comment, marked as answer]
@cx-mezz there's an issue planned for 10.1 release to fix this problem. please see https://github.com/pimcore/pimcore/issues/7187

**Reply by cx-mezz** [reply]
Thanks!
